src/routes/solve/[crossword]/[crossword_id]/[solve_group]/+page.svelte

```diff
@@ -52,19 +52,23 @@
     return allClues.filter((clue: Clue) => clueCoversCell(clue, row, col));
   };
 
-  const setActiveCell = (row: number, col: number) => {
+  const setActiveCell = (row: number, col: number, keepActiveClue: boolean) => {
     if (!networkClues) {
       return;
     }
-    let potential = cluesCoveringCell(row, col);
     activeRowIndex = row;
     activeCellIndex = col;
+    // If we are within the same clue, we're finished.
+    if (keepActiveClue) {
```

Comment:
Hmm would it be possible to make this less imperative? I.e. you just have some 'cursor' state which contains `row`, `column` and `direction` and this is sufficient to figure out the clue always. So you wouldn't need the extra parameter here and the function could just deal with bounds checking then setting the cursor variables. Then also the current clue could just be reactive variable from the cursor?

Reply:
I don't know if that is possible. This is called in two places - when you click on a square, and when you enter a character and you're not at the end of the word. In the first case you explicitly want to change the direction. In the other case, you never do. 

Unless you meant that the calling code should hand in the new direction? But i don't think they easily have access to the other clues. Could have separate functions for these two cases?

Reply:
I agree that current clue should be reactive variable somehow though, they shouldn't get out of sync. maybe that is possible in some other manner?